src/deltacode/utils.py

```diff
@@ -75,45 +80,50 @@ def update_modified_from_license_info(delta, unique_categories):
-            unique_categories_in_old_file = len(old_categories & unique_categories)
-            # 'Permissive' or 'Public Domain' ==> 'Copyleft Limited' or higher
-            if unique_categories_in_old_file == 0 and category in unique_categories:
-                delta.update(20, category.lower() + ' added')
-            # at least 1 category in the old file was 'Copyleft Limited' or higher ==> 'Copyleft Limited' or higher
-            elif unique_categories_in_old_file != 0 and category in unique_categories:
-                delta.update(10, category.lower() + ' added')
-            # 'Permissive' or 'Public Domain' ==> 'Permissive' or 'Public Domain' if not in old_categories
-            elif category not in unique_categories:
-                delta.update(0, category.lower() + ' added')
-
+    try:
+
+        if not delta.new_file.licenses and delta.old_file.licenses:
+            delta.update(15, 'license info removed')
+            return
+
+        new_licenses = delta.new_file.licenses or []
+        old_licenses = delta.old_file.licenses or []
+
+        new_categories = set(license.get('category','') for license in new_licenses)
+        old_categories = set(license.get('category','') for license in old_licenses)
+
+        if delta.new_file.licenses and not delta.old_file.licenses:
+            delta.update(20, 'license info added')
+
+            for category in new_categories:
+                # no license ==> 'Copyleft Limited'or higher
+                if category in unique_categories:
+                    delta.update(20, category.lower() + ' added')
+                # no license ==> 'Permissive' or 'Public Domain'
+                else:
+                    delta.update(0, category.lower() + ' added')
+            return
+
+        new_keys = set(license['key'] for license in delta.new_file.licenses)
+        old_keys = set(license['key'] for license in delta.old_file.licenses)
+
+        if new_keys != old_keys:
+
+            delta.update(10, 'license change')
+            for category in new_categories - old_categories:
+                unique_categories_in_old_file = len(old_categories & unique_categories)
+                # 'Permissive' or 'Public Domain' ==> 'Copyleft Limited' or higher
+                if unique_categories_in_old_file == 0 and category in unique_categories:
+                    delta.update(20, category.lower() + ' added')
+                # at least 1 category in the old file was 'Copyleft Limited' or higher ==> 'Copyleft Limited' or higher
+                elif unique_categories_in_old_file != 0 and category in unique_categories:
+                    delta.update(10, category.lower() + ' added')
+                # 'Permissive' or 'Public Domain' ==> 'Permissive' or 'Public Domain' if not in old_categories
+                elif category not in unique_categories:
+                    delta.update(0, category.lower() + ' added')
+    except AttributeError:
```

Comment:
See my comment above. This whole block is really ugly. I think we are better off ignoring everything having to do with license and copyright checks for now. We can add this later in a much more clean way.